BUILDING.md

```diff
@@ -160,17 +189,25 @@ installed to be able to compile the C/C++ sources.
 > cmake --build .
 ```
 
-Using gRPC C++ as a DLL is not recommended, but you can still enable it by running `cmake` with `-DBUILD_SHARED_LIBS=ON`.
+### Windows Build Notes 
 
-### Windows: A note on building shared libs (DLLs)
+**Building shared libs (DLLs)**
 
-Windows DLL build is supported at a "best effort" basis and we don't recommend using gRPC C++ as a DLL as there are some known drawbacks around how C++ DLLs work on Windows. For example, there is no stable C++ ABI and you can't safely allocate memory in one DLL, and free it in another etc.
+> **Warning**
+> Using gRPC C++ as a DLL is not recommended, but you can still enable it by running `cmake` with `-DBUILD_SHARED_LIBS=ON`. 
+
+Windows DLL build is supported on a "best effort" basis and we don't recommend using gRPC C++ as a DLL as there are some known drawbacks around how C++ DLLs work on Windows. For example, there is no stable C++ ABI and you can't safely allocate memory in one DLL, and free it in another etc.
 
 That said, we don't actively prohibit building DLLs on windows (it can be enabled in cmake with `-DBUILD_SHARED_LIBS=ON`), and are free to use the DLL builds
 at your own risk.
 - you've been warned that there are some important drawbacks and some things might not work at all or will be broken in interesting ways.
 - we don't have extensive testing for DLL builds in place (to avoid maintenance costs, increased test duration etc.) so regressions / build breakages might occur
 
+
+**Debug/Release Installs**
+
+There are both Debug and Release builds and their respective installs differ slightly (e.g. debug/release verions of libs). If you want a 'complete' install that includes both Debug and Release artefacts, then you'll need to perform two builds to separate install target-locations and then merge the differences. 
```

Comment:
Isn't this general Visual Studio usage? I'm not sure this belongs here.

Reply:
No, this is not specific to building using Visual Studio and is also the case when, say, building with CMake. Technically, this is the case across the board (and even across OS) but arguably a more 'sensitive' issue on Windows where you get all sorts of wonderful linker errors when attempting to mix-and-match. 

I wanted to call this out because I had to learn this the hard way. (I might expand the point in a follow-on PR.)

Reply:
I also don't see the usefulness of this section. Generally, we are trying to keep the docs as short as possible which containing the most important information (that's good for both maintainability and readability). TBH, I don't see why I would ever install both debug and opt version of the library and I even don't see how to build both at the same time.

Reply:
The reason why I explicitly called this out is because I stumbled upon a slight inconsistency between the [gRPC Quick start](https://grpc.io/docs/languages/cpp/quickstart/) page and these instructions, which are referenced from there. The build instructions for macOS and Linux both here on this page and on the gRPC Quick start page do not specify a build type. However, the Windows build instructions explicitly specifies Release builds.  (Note that the gRPC Quick start page does not include instructions for Windows.) 

There is a small anomaly between the Debug and Release builds for Windows in that the libs libprotobuf, libprotobuf-lite and libprotoc as well as zlib and zlibstatic are postfixed with a 'd' in the debug build. Obviously, anyone considering using gRPC as a cross-platform IPC mechanism will need to be aware of this so they can reflect it in their own build system. 

I get that we want to keep the docs short but this issue tripped me up so I felt it important to share. Perhaps there is a better place for it?